storage/src/main/java/org/apache/uniffle/storage/handler/impl/ComposedClientReadHandler.java

```diff
@@ -190,71 +120,46 @@ public void updateConsumedBlockInfo(BufferSegment bs, boolean isSkippedMetrics)
     LOG.info(getReadUncompressLengthInfo());
   }
 
   @VisibleForTesting
-  public String getReadBlokNumInfo() {
-    return "Client read " + readHandlerMetric.getReadBlockNum()
-        + " blocks from [" + serverInfo + "], Consumed["
-        + " hot:" + hostHandlerMetric.getReadBlockNum()
-        + " warm:" + warmHandlerMetric.getReadBlockNum()
-        + " cold:" + coldHandlerMetric.getReadBlockNum()
-        + " frozen:" + frozenHandlerMetric.getReadBlockNum()
-        + " ], Skipped[" + " hot:" + hostHandlerMetric.getSkippedReadBlockNum()
-        + " warm:" + warmHandlerMetric.getSkippedReadBlockNum()
-        + " cold:" + coldHandlerMetric.getSkippedReadBlockNum()
-        + " frozen:" + frozenHandlerMetric.getSkippedReadBlockNum() + " ]";
+  public String getReadBlockNumInfo() {
+    return getMetricsInfo("blocks", ClientReadHandlerMetric::getReadBlockNum,
+        ClientReadHandlerMetric::getSkippedReadBlockNum);
   }
 
   @VisibleForTesting
   public String getReadLengthInfo() {
-    return "Client read " + readHandlerMetric.getReadLength()
-        + " bytes from [" + serverInfo + "], Consumed["
-        + " hot:" + hostHandlerMetric.getReadLength()
-        + " warm:" + warmHandlerMetric.getReadLength()
-        + " cold:" + coldHandlerMetric.getReadLength()
-        + " frozen:" + frozenHandlerMetric.getReadLength() + " ], Skipped["
-        + " hot:" + hostHandlerMetric.getSkippedReadLength()
-        + " warm:" + warmHandlerMetric.getSkippedReadLength()
-        + " cold:" + coldHandlerMetric.getSkippedReadLength()
-        + " frozen:" + frozenHandlerMetric.getSkippedReadLength() + " ]";
+    return getMetricsInfo("bytes", ClientReadHandlerMetric::getReadLength,
+        ClientReadHandlerMetric::getSkippedReadLength);
   }
 
   @VisibleForTesting
   public String getReadUncompressLengthInfo() {
-    return "Client read " + readHandlerMetric.getReadUncompressLength()
-        + " uncompressed bytes from [" + serverInfo + "], Consumed["
-        + " hot:" + hostHandlerMetric.getReadUncompressLength()
-        + " warm:" + warmHandlerMetric.getReadUncompressLength()
-        + " cold:" + coldHandlerMetric.getReadUncompressLength()
-        + " frozen:" + frozenHandlerMetric.getReadUncompressLength() + " ], Skipped["
-        + " hot:" + hostHandlerMetric.getSkippedReadUncompressLength()
-        + " warm:" + warmHandlerMetric.getSkippedReadUncompressLength()
-        + " cold:" + coldHandlerMetric.getSkippedReadUncompressLength()
-        + " frozen:" + frozenHandlerMetric.getSkippedReadUncompressLength() + " ]";
+    return getMetricsInfo("uncompressed bytes", ClientReadHandlerMetric::getReadUncompressLength,
+        ClientReadHandlerMetric::getSkippedReadUncompressLength);
+  }
+
+  private String getMetricsInfo(String name, Function<ClientReadHandlerMetric, Long> consumed,
+      Function<ClientReadHandlerMetric, Long> skipped) {
+    return "Client read " + consumed.apply(readHandlerMetric)
```

Comment:
Could you change it to StringBuilder by the way?

Reply:
OK. `StringBuilder` is less intuitive though.